config/setup/git.go

```diff
@@ -20,13 +21,26 @@ func Git(c *Controller) (middleware.Middleware, error) {
 		return nil, err
 	}
 
-	c.Startup = append(c.Startup, func() error {
-		// Start service routine in background
-		git.Start(repo)
+	// If a HookUrl is set, we switch to event based pulling.
+	// Install the url handler
+	if repo.HookUrl != "" {
 
-		// Do a pull right away to return error
-		return repo.Pull()
-	})
+		webhook := &webhook.WebHook{Repo: repo}
+		return func(next middleware.Handler) middleware.Handler {
+			webhook.Next = next
+			return webhook
+		}, nil
+
```

Comment:
Kind of... :) Could you clarify?

Something else to note, about this PR currently: If the user pulls at intervals, the repo will pull down immediately at server start. If the user uses a webhook, the behavior is different: the site does *not* get cloned down when the server starts, possibly leaving the site root empty.

I think the behavior should be consistent, whichever method is used.